### PR TITLE
filename filter maps

### PR DESCRIPTION
**Pull request by 1-alex98**

### CODE REVIEW

**Comment by axel1200** [comment]
How do we go about it?

**Comment by bukajsytlos** [comment]
Maybe you could use FileNameUtil. If you have windows environment you could run tests. @micheljung was complaining about test failing under windows. But it is working fine in linux env. Maybe regex can be tweaked to support both platforms' filenames

**Comment by axel1200** [comment]
Where should I use FileNameUtil?

**Comment by axel1200** [comment]
Can you comment on lines ?

**Comment by axel1200** [comment]
The approach here is to just remove faulty characters here on upload

File: src/main/java/com/faforever/api/map/MapService.java [hunk 1]
```diff
@@ -71,6 +71,8 @@ void uploadMap(byte[] mapData, String mapFilename, Player author, boolean isRank
     Assert.notNull(author, "'author' must not be null");
     Assert.isTrue(mapData.length > 0, "'mapData' must not be empty");
 
+    mapFilename = mapFilename.replaceAll("[^\\w.\\-]", ""); //replacing all characters that are illegal in filenames
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/map/MapService.java]: consider using FileNameUtil.normalizeFileName method

**Comment by axel1200** [comment]
I did

**Comment by Brutus5000** [comment]
Due to current cases on prod we not only need to check the file name but also remove all non-ascii characters from the map name itself. @axel1200 Could you add this here?

**Comment by axel1200** [comment]
Ehmm is that not what I was doing?

**Comment by axel1200** [comment]
@bukajsytlos I looked into the normalize file name method and it is pretty bad it replaces %| and some other character instead of replacing everything that is not a letter or an underscore. because I could imagine some additional characters that wont get replaced like this and would cause problems again...

**Comment by bukajsytlos** [comment]
@axel1200 yeah I take a look at it too. The thing is that it should produce valid filenames from OS stand-point of view. If this does not meet business requirement for map names we could create new name transformation. Not sure which characters should be removed for backward compatibility

**Comment by axel1200** [comment]
tab for example should be removed

**Comment by 1-alex98** [comment]
for example today somebody uploaded map with : character in it and broke the vault

**Comment by Brutus5000** [comment]
Short update: I am currently working on improvements of this PR

**Comment by 1-alex98** [comment]
Do so

**Pull request merged**